bib/cmd/bootc-image-builder/main.go

```diff
@@ -176,11 +176,15 @@ func saveManifest(ms manifest.OSBuildManifest, fpath string) error {
 	if err != nil {
 		return fmt.Errorf("failed to create output file %q: %s", fpath, err.Error())
 	}
-	// nolint:errcheck
-	defer fp.Close()
+	defer func() {
+		if err = fp.Close(); err != nil {
+			err = fmt.Errorf("failed to close output file %q: %s", fpath, err.Error())
```

Comment:
We need to keep the error with `%w` here, but lets just not do it and use either `errors.Join()` or add  `return fp.Close()` at the end.

Reply:
I think in case of other errors the defer is a good Idea?
I'll adapt and propose your short version, which is a good "pattern" I guess.